medusa/clients/torrent/generic.py

```diff
@@ -232,66 +202,47 @@ def send_torrent(self, result):
         :return:
         :rtype: str or bool
         """
-        r_code = False
 
-        log.debug('Calling {name} Client', {'name': self.name})
 
-        if not self.auth:
-            if not self._get_auth():
-                log.warning('{name}: Authentication Failed', {'name': self.name})
-                return r_code
+        # Sets per provider seed ratio
+        result.ratio = result.provider.seed_ratio()
 
+        # check for the hash and add it if not there
         try:
-            # Sets per provider seed ratio
-            result.ratio = result.provider.seed_ratio()
-
-            # lazy fix for now, I'm sure we already do this somewhere else too
-            result = self._get_info_hash(result)
-
             if not result.hash:
+                raise Exception()
```

Comment:
I don't think you need to use exceptions here. Can just do
```
if not result.hash and not self._get_info_hash(result):
    return False
```

Reply:
If the result.hash is not yet defined then this exception is needed otherwise it will kill this function  by means of an unhandled exception

Reply:
result is a SearchResult object. `https://github.com/pymedusa/Medusa/blob/master/medusa/classes.py#L79`.
So in "worst case" it's None.